url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/examples/Joust.java

```diff
@@ -0,0 +1,419 @@
+package fri.patterns.interpreter.parsergenerator.examples;
+import java.util.*;
+
+import fri.patterns.interpreter.parsergenerator.*;
+import fri.patterns.interpreter.parsergenerator.semantics.*;
+//import fri.patterns.interpreter.parsergenerator.builder.SerializedParser;
+import fri.patterns.interpreter.parsergenerator.parsertables.LALRParserTables;
+import fri.patterns.interpreter.parsergenerator.lexer.LexerBuilder;
+import fri.patterns.interpreter.parsergenerator.syntax.builder.SyntaxSeparation;
+import fri.patterns.interpreter.parsergenerator.syntax.Syntax;
+
+/**
+	Calculator for arithmetic expressions, showing the elegance of ReflectSemantic.
+	<p>
+	Syntax: java fri.patterns.interpreter.parsergenerator.examples.Calculator '(4+2.3) *(2 - -6) + 3*2'
+	
+*/
+
+public class Joust extends ReflectSemantic
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class contains 19 clusters of instance methods that do not have any accessed class members in common. For example, a cluster might have 2 methods that access only the class fields x and y, and another cluster might have 2 other methods that access only the class fields a and b. By comparison, 99% of the classes in the CodeGuru reference dataset contain fewer clusters. A high number of these clusters indicates low class cohesion. Classes with low class cohesion contain unrelated operations which make them difficult to understand and less likely to be used. We recommend that you simplify this class or break it into multiple cohesive classes.